### PR TITLE
fix: Optimize propagateSelected by limiting matching items to the selected ones (#2929)

### PR DESCRIPTION
**Pull request by koenvd**
Closes carbon-design-system/carbon-components-angular#2929

Optimizes `propogateSelected` as in the first bullet point explained in issue #2929.  

Ideally we still want to improve this doing a lookup with a specific key instead of looping over all items.  But as a start we only loop over selected items which makes the performance already a lot better since typically the amount of selected items is way less then the total number of items.

Since this my first MR to this project please let me know if I need to add tests or do something extra to get this one merged.

Thanks.

### CODE REVIEW

**Comment by Akshat55** [comment]
@koenvd Could you please sign the DCO, simply comment the following text to the PR:

I have read the DCO document and I hereby sign the DCO.

**Comment by koenvd** [comment]
I have read the DCO document and I hereby sign the DCO.

**Comment by koenvd** [comment]
Done @Akshat55 .

**Review by Akshat55 — APPROVED** [review]

**Review by zvonimirfras — APPROVED** [review]

File: src/dropdown/list/dropdown-list.component.ts [hunk 1]
```diff
@@ -445,24 +445,38 @@ export class DropdownList implements AbstractDropdownView, AfterViewInit, OnDest
...
 				// stringify for compare
 				tempOldItem = JSON.stringify(tempOldItem);
-				for (let newItem of value) {
-					// copy the item
-					let tempNewItem: string | ListItem = Object.assign({}, newItem);
-					// deleted selected because it's what we _want_ to change
-					delete tempNewItem.selected;
-					// stringify for compare
-					tempNewItem = JSON.stringify(tempNewItem);
+				for (let selectedNewItem of selectedNewItems) {
```
Comment (Licen-it) [inline src/dropdown/list/dropdown-list.component.ts:476]: i'm arriving late to the party, but here we could optimise even more as follows (even in terms of readability 😄 )

```ts
oldItem.selected = !!selectedNewItems.find(newItem => tempOldItem.includes(newItem))
```
avoiding the entire for block (for sure `.find` is better optimised than a for of...and we would avoid reassigning `oldItem.selected` to false multiple time when the array is relatively big

**Pull request merged**